systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java

```diff
@@ -351,33 +352,41 @@ void testCustomBridgeLabelsAreProperlySet(ExtensionContext extensionContext) {
 
     @BeforeAll
     void createClassResources(ExtensionContext extensionContext) {
-        prepareEnvForOperator(extensionContext, NAMESPACE);
+        install.prepareEnvForOperator(extensionContext, NAMESPACE);
         // Apply role bindings in CO namespace
-        applyBindings(extensionContext, NAMESPACE);
+        Install.applyBindings(extensionContext, NAMESPACE);
 
         // Create ClusterRoleBindings that grant cluster-wide access to all OpenShift projects
         List<ClusterRoleBinding> clusterRoleBindingList = ClusterRoleBindingTemplates.clusterRoleBindingsForAllNamespaces(NAMESPACE);
         clusterRoleBindingList.forEach(clusterRoleBinding ->
             ClusterRoleBindingResource.clusterRoleBinding(extensionContext, clusterRoleBinding));
         // deploy CO with service labels and annotations envs
-        resourceManager.createResource(extensionContext, BundleResource.clusterOperator(NAMESPACE, NAMESPACE, Constants.CO_OPERATION_TIMEOUT_DEFAULT, Constants.RECONCILIATION_INTERVAL)
-            .editSpec()
-                .editTemplate()
-                    .editSpec()
-                        .editFirstContainer()
-                            .addNewEnv()
-                                .withName("STRIMZI_CUSTOM_KAFKA_BRIDGE_SERVICE_LABELS")
-                                .withValue("app=bar")
-                            .endEnv()
-                            .addNewEnv()
-                                .withName("STRIMZI_CUSTOM_KAFKA_BRIDGE_SERVICE_ANNOTATIONS")
-                                .withValue("bar=app")
-                            .endEnv()
-                        .endContainer()
-                    .endSpec()
-                .endTemplate()
-            .endSpec()
-            .build());
+        resourceManager.createResource(extensionContext,
+            new BundleResource.BundleResourceBuilder()
```

Comment:
Just thinking - would it be possible to add to the `Install` class something like `withType()` where you specify the `Bundle`,`Helm` or `OLM` and then you proceed with all other steps? Maybe it will be complicated, but most of the parameters are same. WDYT @Frawless @see-quick ?

Reply:
So you suggest one builder based on the type from `Install`?

Reply:
Yeah, something like that.